pkg/roachprod/prometheus/prometheus.go

```diff
@@ -0,0 +1,487 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package prometheus
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/roachprod/config"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/install"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/logger"
+	"github.com/cockroachdb/errors"
+	promv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+	"gopkg.in/yaml.v2"
+)
+
+const workloadPrometheusPort = 2112
+
+// Client is an interface allowing queries against Prometheus.
+type Client interface {
+	Query(ctx context.Context, query string, ts time.Time) (model.Value, promv1.Warnings, error)
+}
+
+// ScrapeNode is a node to scrape from.
+type ScrapeNode struct {
+	Nodes install.Nodes
+	IPs   []string
+	Port  int
+}
+
+// ScrapeConfig represents a single instance of scraping.
+// Note how workload scrapes are set up differently than CRDB binary scrapes.
+type ScrapeConfig struct {
+	JobName     string
+	MetricsPath string
+	ScrapeNodes []ScrapeNode
+	Labels      map[string]string // additional static labels to add
+}
+
+// Config is a monitor that watches over the running of prometheus.
+type Config struct {
+	// PrometheusNode identifies a single node in the cluster to run the prometheus instance on.
+	// The type is install.Nodes merely for ease of use.
+	PrometheusNode install.Nodes
+
+	// ScrapeConfigs provides the configurations for each scraping instance
+	ScrapeConfigs []ScrapeConfig
+
+	// NodeExporter identifies each node in the cluster to scrape
+	NodeExporter install.Nodes
+
+	// NodeExporter URLs
+	NodeExporterIPs []string
+
+	// Grafana provides the info to set up grafana
+	Grafana GrafanaConfig
+}
+
+// GrafanaConfig are options related to setting up a Grafana instance.
+type GrafanaConfig struct {
+	Enabled bool
```

Comment:
When starting the grafana server, we don't seem to be setting this to true, which I think we want.

Reply:
See discussion above, I don't think the emphasis on prometheus with an optional grafana server is quite right for this command.

Reply:
Cool. StartGrafana() now sets this to true.

Reply:
Done.